### PR TITLE
OTLP+Arrow receiver skeleton

### PR DESCRIPTION
**Pull request by jmacd**
**Description:** 
Uses the latest `otel-arrow-adapter` producer APIs for spans.

**Testing:**
This builds and existing tests continue to pass.  No new testing.

### CODE REVIEW

**Comment by jmacd** [comment]
Note that most of the work in getting to this first milestone was in preparing the `otel-arrow-adapter`. The four prerequisites are: https://github.com/lquerel/otel-arrow-adapter/pulls?q=is%3Apr+is%3Aclosed+author%3Ajmacd+is%3Amerged

File: receiver/otlpreceiver/internal/arrow/arrow.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+				case arrowpb.OtlpArrowPayloadType_METRICS:
+					err = fmt.Errorf("no metrics consumer")
+				case arrowpb.OtlpArrowPayloadType_LOGS:
+					err = fmt.Errorf("no logs consumer")
+				case arrowpb.OtlpArrowPayloadType_SPANS:
+					if traceProducer == nil {
+						err = fmt.Errorf("no spans consumer")
+						continue
+					}
+					var otlp []ptrace.Traces
```
Comment (jaronoff97) [inline receiver/otlpreceiver/internal/arrow/arrow.go:94]: I don't think we need to declare the slice here

File: receiver/otlpreceiver/internal/arrow/arrow.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		if err != nil {
+			return err
+		}
+
+		messages, err := r.arrowConsumer.Consume(req)
+
+		status := &arrowpb.StatusMessage{
+			BatchId: req.GetBatchId(),
+		}
+		if err == nil {
```
Comment (jaronoff97) [inline receiver/otlpreceiver/internal/arrow/arrow.go:81]: Should we invert the condition so the main logic isn't in the if? The readability of two if conditions for the same err == nil is a bit odd.

**Review by jaronoff97 — COMMENTED** [review]
a few thoughts, main one being some logic changes in the events stream handler

**Review by lquerel — APPROVED** [review]

File: receiver/otlpreceiver/config.go [hunk 3]
```diff
@@ -52,6 +58,9 @@ func (cfg *Config) Validate() error {
 		cfg.HTTP == nil {
 		return errors.New("must specify at least one protocol when using the OTLP receiver")
 	}
+	if cfg.Arrow != nil && cfg.Arrow.Enabled && cfg.GRPC == nil {
```
[the remark this answers is not in the document]
Comment (lquerel) [reply receiver/otlpreceiver/config.go:61]: gRPC stream ensures that the same backend/collector instance will receive consecutive messages for the same stream. Without a stream semantic and with a load balancer in front of the backend/collector there is no such guarantee for a standard rpc call.

**Pull request merged**